File: 3217 Delete Nodes From Linked List Present in Array/3217.cpp
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        ListNode *ans = new ListNode(0, head), *prev = ans, *cur = prev->next;
        unordered_set<int> invalid;
        for (int num: nums) invalid.insert(num);
        while (cur != nullptr){
            if (invalid.count(cur->val)) prev->next = cur->next; 
            else prev = prev->next;
            cur = cur->next;
        }
        return ans->next;
    }
};

File: 3217 Delete Nodes From Linked List Present in Array/3217.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        ans = p = c = ListNode(0, head); bad = set(nums)
        while c := c.next : p.next, p = (c.next, p) if c.val in bad else (p.next, p.next)
        return ans.next
